### DISCUSSION TITLE
A HTTP client (like postman) built in terminal? (just an proposal and collect ideas, not done yet)

### DISCUSSION THREAD

**Discussion by laixintao** [Textualize/textual#3664] [Show and tell]
Hi there,

I am not sure if this is the right place to post this cause I haven't done anything yet, if this is improper, you can move it.

Years ago I use postman for send HTTP requests for testing, developing, but postman get more and more complicated then I found insomnia, it's simple, clean, and easy to use.

Then years later insomnia sold to Kong, and got more and more complicated and buggy, start to selling services. I don't think they put much efforts on client side anymore, they more want to just sell cloud services:

1. you need a account to use this client
2. the update alert will pop up 100 times a day (I believe it's a bug)
3. and other bugs (but to be fair, no fatal bug, just small issues, I can still use it)

...which makes me think if I should write an HTTP client based on TUI myself.

It:

1. should be easy to use
2. Can just integrate with `HAR` files, like this: `textual-http-client foo.har` then you enter the TUI, with that, users can manage their HARs files whatever they want, like put all of the files in the project file, then goes with the project. Or a team can put all HARs into a git then work on git. (Wow, did a just design Insomnia's "cloud service"?)
3. Can work with `curl`, for example, if someone send you a `curl` and you want to modify it, you can paste the curl command into terminal then press a shortcut key or ` echo "curl ....." | textual-http-client` then enter the TUI
4. Those above is something GUI can not do (or not easy to do), and of course, most feature and postman and insomnia do, we need to have it too.

With the awesome textual I think I can archive almost the same UI experience like electron do.

Any feedbacks or ideas? Do I miss something? Or was there already a TUI tool?

**Comment by davep** [comment]
> Or was there already a TUI tool?

I'm aware of at least [http-tui](https://github.com/treyhunner/http-tui).

**Reply by laixintao** [reply]
Thanks, definitely will try it out!

**Comment by willmcgugan** [comment]
Sounds like useful thing. I could see myself using it.

**Comment by ManuelSchenk** [comment]
Have you checked: https://posting.sh/
